### PR TITLE
[improvement]Use phmap for aggregation with integer keys

### PR DESCRIPTION
**Pull request by mrhhsg**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

After some testing on clickbench, it was proved phmap has a performance improvement for aggregation with numeric key.

clickhouse bench q16: 4.38s--->3.81s; q33: 37s--->27s

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wangbo** [comment]
Can string type in serialized key can use this optimization?

File: be/src/vec/exec/vaggregation_node.h [hunk 1]
```diff
@@ -157,7 +157,7 @@ struct AggregationMethodStringNoCache {
 
 /// For the case where there is one numeric key.
 /// FieldType is UInt8/16/32/64 for any type with corresponding bit width.
-template <typename FieldType, typename TData, bool consecutive_keys_optimization = true>
+template <typename FieldType, typename TData, bool consecutive_keys_optimization = false>
```
Comment (wangbo) [inline be/src/vec/exec/vaggregation_node.h:160]: Why set consecutive_keys_optimization = false here?

File: be/src/vec/exec/vaggregation_node.cpp [hunk 2]
```diff
@@ -208,20 +229,36 @@ void AggregationNode::_init_hash_method(std::vector<VExprContext*>& probe_exprs)
         if (use_fixed_key) {
             if (has_null) {
                 if (std::tuple_size<KeysNullMap<UInt64>>::value + key_byte_size <= sizeof(UInt64)) {
-                    _agg_data.init(AggregatedDataVariants::Type::int64_keys, has_null);
+                    if (_is_streaming_preagg)
+                        _agg_data.init(AggregatedDataVariants::Type::int64_keys, has_null);
+                    else
+                        _agg_data.init(AggregatedDataVariants::Type::int64_keys_phase2, has_null);
```
Comment (wangbo) [inline be/src/vec/exec/vaggregation_node.cpp:235]: if _is_streaming_preagg == false, this  means current AggNode maybe the second phase of a streaming agg, but also may be a first phase of BlockAgg.

Comment (mrhhsg) [reply be/src/vec/exec/vaggregation_node.cpp:233]: Yes, here replaced this with `_is_merge` to check.

File: be/src/vec/exec/vaggregation_node.h (hunk 1 above)
Comment (mrhhsg) [reply be/src/vec/exec/vaggregation_node.h:160]: Restricted to phmap's value_type contains const field(key), the cached value cannot be updated.

File: be/src/vec/exec/vaggregation_node.h [hunk 3]
```diff
@@ -356,46 +356,75 @@ struct AggregationMethodSingleNullableColumn : public SingleColumnMethod {
...
-using AggregatedDataWithUInt128Key = HashMap<UInt128, AggregateDataPtr, HashCRC32<UInt128>>;
-using AggregatedDataWithUInt256Key = HashMap<UInt256, AggregateDataPtr, HashCRC32<UInt256>>;
+using AggregatedDataWithUInt32Key = PHHashMap<UInt32, AggregateDataPtr, HashCRC32<UInt32>>;
+using AggregatedDataWithUInt64Key = PHHashMap<UInt64, AggregateDataPtr, HashCRC32<UInt64>>;
+using AggregatedDataWithUInt128Key = PHHashMap<UInt128, AggregateDataPtr, HashCRC32<UInt128>>;
+using AggregatedDataWithUInt256Key = PHHashMap<UInt256, AggregateDataPtr, HashCRC32<UInt256>>;
+using AggregatedDataWithUInt32KeyPhase2 =
+        PHHashMap<UInt32, AggregateDataPtr, HashMixWrapper<UInt32>>;
+using AggregatedDataWithUInt64KeyPhase2 =
+        PHHashMap<UInt64, AggregateDataPtr, HashMixWrapper<UInt64>>;
```
Comment (Gabriel39) [inline be/src/vec/exec/vaggregation_node.h:366]: What's different between `HashMixWrapper` and a pure `HashCRC32`?

File: be/src/vec/exec/vaggregation_node.cpp [hunk 4]
```diff
@@ -1155,10 +1226,13 @@ Status AggregationNode::_merge_with_serialized_key(Block* block) {
                     auto emplace_result = [&]() {
                         if constexpr (IsPhmapTraits<HashTableType>::value) {
                             if (LIKELY(i + HASH_MAP_PREFETCH_DIST < rows)) {
-                                agg_method.data.prefetch_by_hash(
-                                        hash_values[i + HASH_MAP_PREFETCH_DIST]);
+                                if constexpr (IsParallelPhmapTraits<HashTableType>::value) {
```
Comment (Gabriel39) [inline be/src/vec/exec/vaggregation_node.cpp:1229]: FYI, maybe we can wrap `IsParallelPhmapTraits` and `IsPreSerializedKeysHashMethodTraits` into one single Traits? Using `IsPreSerializedKeysHashMethodTraits` above but `IsParallelPhmapTraits` here is a bit confused

File: be/src/vec/exec/vaggregation_node.h (hunk 3 above)
Comment (mrhhsg) [reply be/src/vec/exec/vaggregation_node.h:366]: Phmap has performance issues with phase 2 of aggregation if the hash algorithm is the same as phase 1.

File: be/src/vec/exec/vaggregation_node.cpp (hunk 4 above)
Comment (mrhhsg) [reply be/src/vec/exec/vaggregation_node.cpp:1229]: `IsPreSerializedKeysHashMethodTraits` is about `aggregate state`
I will combine `IsParallelPhmapTraits` and `IsPhmapTraits` in one traits.

**Review by wangbo — APPROVED** [review]
LGTM

**Review by kpfly — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**